lib/common/client/middleware/response/snakecase.rb

```diff
@@ -10,21 +10,26 @@ def initialize(app, options = { symbolize: true })
           end
 
           def on_complete(env)
-            return unless env.response_headers['content-type'] =~ /\bjson/
+            # return false unless env.response_headers['content-type'] =~ /\b(xml|json)/
+            return unless deserialized_body?(env.body)
             env.body = parse(env.body)
           end
 
-          def parse(parsed_json)
-            case parsed_json
+          def parse(parsed_object)
+            case parsed_object
             when Array
-              parsed_json.map { |hash| transform(hash) }
+              parsed_object.map { |hash| transform(hash) }
             when Hash
-              transform(parsed_json)
+              transform(parsed_object)
```

Comment:
Note: I can simplify this logic by always calling parse and putting an `else` in the case statement. Just wanted to keep changes to a minimum to help draw attention to possible implications.